### PR TITLE
8241865: sun/management/jdp tests fail (Timed out waiting for JDP packet) (macosx-aarch64)

### PR DESCRIPTION
**Pull request by kevinjwalls**
This test has been noisy lately, and all the failures have been on "Mac OS X 15.0.1 (aarch64)".

Forcing it to require any other OS version looks good in my testing.

---------
### Progress
- [ ] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8241865](https://bugs.openjdk.org/browse/JDK-8241865): sun/management/jdp tests fail (Timed out waiting for JDP packet) (macosx-aarch64) (**Bug** - P4)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/22742/head:pull/22742` \
`$ git checkout pull/22742`

Update a local copy of the PR: \
`$ git checkout pull/22742` \
`$ git pull https://git.openjdk.org/jdk.git pull/22742/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 22742`

View PR using the GUI difftool: \
`$ git pr show -t 22742`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/22742.diff">https://git.openjdk.org/jdk/pull/22742.diff</a>

</details>
<details><summary>Using Webrev</summary>

[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/22742#issuecomment-2542183621)
</details>

### CODE REVIEW

**Comment by kevinjwalls** [comment]
There was a Linux x64 failure a few years back, but since un-problemlisting in August 2024, macos aarch64 is where we are seeing failures.  A batch of 20 run will generally see a few failures on that platform and nowhere else.

I did some experiments with other fixes such as long timeouts, retrying the test, pausing and retrying the test.
But the OSX 15.0.1 failure is not solved by any of these.  Requiring any other OS version seems better than problemlisting, and means we do run this test on multiple other MacOSX aarch64 versions.

While here:

Some typos and formatting changes.

When handling SocketTimeoutException, don't capitalise the method as "onSocketTimeOut()", make the "Timeout" case consistent (which made my searching easier).

Remove the ' quotes from the loggers, that looks strange in the logs, probably unintentional.  Output is now like:
2024-12-12 18:39:50 FINE    Pause in between packets is: 1 seconds.
2024-12-12 18:39:50 FINE    Timeout set to 20 seconds.

File: test/jdk/sun/management/jdp/JdpDefaultsTest.java [hunk 1]
```diff
@@ -30,7 +30,7 @@
 /*
  * @test JdpDefaultsTest
  * @summary Assert that we can read JDP packets from a multicast socket connection, on default IP and port.
- *
+ * @requires os.version != "Mac OS X 15.0.1 (aarch64)"
```
Comment (plummercj) [inline test/jdk/sun/management/jdp/JdpDefaultsTest.java:33]: Unless there is a known reason why this test cannot run on this platform, this is probably not the appropriate fix here. I'd suggest keeping the CR open and problem listing for macosx-aarch64.

Comment (kevinjwalls) [reply test/jdk/sun/management/jdp/JdpDefaultsTest.java:33]: Ok, I was thinking about debating that after many failures on 15.0.1 and nothing else... But just now we see one on "Mac OS X 12.7.1 (aarch64)" so problemlisting for macosx-aarch64 generally now seems right! (will update)

**Comment by kevinjwalls** [comment]
Closing this and following up with:

Problemlisting: https://github.com/openjdk/jdk/pull/22758

Other JDP test tidyups: https://github.com/openjdk/jdk/pull/22761

**Pull request closed without merging**